src/openms/source/SYSTEM/File.cpp

```diff
@@ -140,6 +140,54 @@ namespace OpenMS
     return true;
   }
 
+  // https://stackoverflow.com/questions/2536524/copy-directory-using-qt
+  bool File::copyDirRecursively(const QString &fromDir, const QString &toDir, bool overwrite_existing)
+  {
+    if (fromDir == toDir)
```

Comment:
you might want to use `QString QDir::canonicalPath()` here, since providing
`fromDir = /tmp/myPath/../` and ` toDir = /tmp/` is actually the same path, but simple string comparison would not detect that.